### DISCUSSION TITLE
Bot getting rate limited randomly?

### DISCUSSION THREAD

**Discussion by Juustar** [jagrosh/MusicBot#1445] [Troubleshooting] [answered]
So I made a bot with this, but it keeps getting a "[WARN] [RateLimiter]" in the console and stops responding to commands for a while? The bot is only in one small server for me and my friends, so I'm not sure why it's getting rate limited

**Comment by Koderew** [comment]
Are you self-hosting?

**Comment by MWE001** [comment, marked as answer]
I don't believe it has anything at all to do with self hosting (not trying to step on any toes here). What I firmly believe is going on is, if we look at the logs in the JMusicBot, this rate limit is happening once every 10 minutes give or take. 

Try this, get rid of the channel description where it updates what song is playing and see if the rate limit goes away. I firmly believe that is our culprit. We knew long ago this was going to be an issue back then and Discord just kicked us in the bean bags once again and lowered our refresh rates even more rendering the topic announcement for "Current Song" Useless at this point. I'm not sure else how to do this but the current way is pretty much now useless.

Try it and see. What do you have to lose right? And if I am wrong, I stand corrected and at least we gave it a try for science. No one else here is trying to discuss it or solve it right?

For the record, I am getting it too. Here is my logs:

 [WARN] [RateLimiter]: Encountered 429 on route PATCH/channels/{channel_id} with bucket .... and a bunch of numbers with channel id's.

**Reply by MWE001** [reply]
And I know this post if from back in October. I have been away for some time and only just realized this was happening to me as well.

**Reply by MichailiK** [reply]
That's correct, JMusicBot trying to update the channel topic & Discord introduced new ratelimits a while back. The topic can now only be updated 2 times every 10 minutes. Because we're planning on releasing 0.4.0 very soon, we've decided to [remove this feature](https://github.com/jagrosh/MusicBot/pull/1485).

**Reply by MWE001** [reply]
Sad news but great news at the same time. We knew long ago it was coming and was going to be an issue as stated above and this day was going to come. It is a cool little feature. Too bad it has to go. I hate having to type a command up to bring a NP playing player type of thing up but it is what it is. We can't control what Discord is going to limit us to. 

Looking forward to 0.4.0 

Yeah, I'm a dork. I just gave myself a thumbs up. SMDH!! That was meant for you Mich.  :-D
